File: 프로그래머스/unrated/181925. 수 조작하기 2/수 조작하기 2.py
def solution(numLog):
    answer = ''
    for i in range(len(numLog)-1):
        if int(numLog[i+1])-int(numLog[i]) == 1:
            answer += 'w'
        elif int(numLog[i+1]) - int(numLog[i]) == -1:
            answer += 's'
        elif int(numLog[i+1]) -int(numLog[i]) == 10:
            answer += 'd'
        elif int(numLog[i+1]) - int(numLog[i]) == -10:
            answer+= 'a'
    return answer

File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.js
function solution(numLog) {
  const convert = {
    1: 'w',
    '-1': 's',
    10: 'd',
    '-10': 'a',
  };

  return numLog
    .map((v, i, arr) => (i === 0 ? '' : convert[v - arr[i - 1]]))
    .join('');
}
